gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -949,18 +964,24 @@ public void run() {
     }
 
     void removeEntry(Map regionEntryMap, RegionEntry re, LocalRegion region) {
-      if (regionEntryMap.remove(re.getKey()) == null) return;
-      if (GemFireCacheImpl.hasNewOffHeap()) {
-        // also remove reference to region buffer, if any
-        Object value = re._getValue();
-        if (value instanceof SerializedDiskBuffer) {
-          ((SerializedDiskBuffer) value).release();
+      Object removedVal = ((CustomEntryConcurrentHashMap)regionEntryMap).remove(re.getKey(),
+          re, entryRemover, null, null);
+
+      if (removedVal == null) return;
+
+      if (removedVal == re) {
```

Comment:
This will always be true with the above remove call which will check for value to be equal to re else will not do the remove. Also two argument remove can be used above as mentioned before.

Reply:
Will remove it. It was for testing purpose.